Subaharini/OOPS/src/oops/LoanManagement.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    //here the compile time polymorphism is used where the calculateInterest contains different parameters... 
+    public void calculateInterest() {
+        System.out.println("Interest for loan is: " + (amount * interest));
+    }
+
+    public void calculateInterest(double newInterest) {
+        System.out.println("Interest for loan with new rate is: " + (amount * newInterest));
+    }
+
+    //abstract method which will be implemented by any subclass of loan class which has its own implementations..
+    public abstract void approveLoan();
+}
+
+//HomeLoan and CarLoan classes uses inheritance for code reusability.. 
+class HomeLoan extends Loan {
+    HomeLoan(double amount, double interest) {
+        super(amount, interest);
+    }
+
+    //method overriding is used here as run time polymorphism
+    //both home loan and car loan has specific implementation method and based on the class the method gets overridden..
+    public void approveLoan() {
+        if (getAmount() > 20000) {
+            System.out.println("Home Loan not approved");
+        } else {
+            System.out.println("Home Loan approved");
+        }
+    }
+}
+
+class CarLoan extends Loan {
+    CarLoan(double amount, double interest) {
+        super(amount, interest);
+    }
+
+    public void approveLoan() {
+        if (getAmount() > 30000) {
+            System.out.println("Car Loan not approved");
+        } else {
+            System.out.println("Car Loan approved");
+        }
+    }
+}
+
+public class LoanManagement {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Enter the amount and interest rate for Home Loan:");
+        double homeLoanAmount = sc.nextDouble();
+        double homeLoanInterest = sc.nextDouble();
+        sc.nextLine();  
+        Loan homeLoan = new HomeLoan(homeLoanAmount, homeLoanInterest);
+        homeLoan.applyForLoan();
+        homeLoan.calculateInterest();
+        System.out.println("Do you need to update interest for Home Loan (Y or N):");
+        String updateInterest = sc.nextLine();
+        if (updateInterest.equalsIgnoreCase("Y")) {
+            System.out.println("Enter new interest rate for Home Loan:");
+            double newInterest = sc.nextDouble();
+            sc.nextLine();  
+            homeLoan.calculateInterest(newInterest);
+        }
+        homeLoan.approveLoan();
+        System.out.println();
+        System.out.println("Enter the amount and interest rate for Car Loan:");
+        double carLoanAmount = sc.nextDouble();
+        double carLoanInterest = sc.nextDouble();
+        sc.nextLine(); 
+        Loan carLoan = new CarLoan(carLoanAmount, carLoanInterest);
+        carLoan.applyForLoan();
+        carLoan.calculateInterest();
+
+        System.out.println("Do you need to update interest for Car Loan (Y or N):");
+        updateInterest = sc.nextLine();
+        if (updateInterest.equalsIgnoreCase("Y")) {
```

Comment:
is it possible to keep this value in Constant ? because if i want to change this in future i should be able to change in one place